test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const openListView = async ( { page, pressKeyWithModifier } ) => {
+	// Open List View with keyboard.
+	await pressKeyWithModifier( 'access', 'o' );
+	// Wait until the current item is selected to ensure UI is ready.
+	await page
+		.getByRole( 'gridcell', {
+			name: /link/i,
+			selected: true,
+		} )
+		.waitFor();
+};
+
+test.describe( 'List View', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'allows a user to drag a block to a new sibling position', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Ensure the setup is correct before dragging.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+				{ name: 'core/paragraph' },
+			] );
+
+		// Drag the paragraph above the heading.
+		const paragraphBlockItem = page.getByRole( 'gridcell', {
+			name: 'Paragraph link',
+		} );
+		const headingBlockItem = page.getByRole( 'gridcell', {
+			name: 'Heading link',
+		} );
+		await paragraphBlockItem.dragTo( headingBlockItem, { x: 0, y: 0 } );
+
+		// Ensure the block was dropped correctly.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/paragraph' },
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+			] );
+	} );
+
+	// Check for regressions of https://github.com/WordPress/gutenberg/issues/38763.
+	test( 'shows the correct amount of blocks after a block is removed in the canvas', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Go to the image block in List View.
+		await pageUtils.pressKeyTimes( 'ArrowUp', 2 );
+		await expect(
+			page
+				.getByRole( 'gridcell', {
+					name: 'Image link',
+				} )
+				.locator( 'a' )
```

Comment:
Could this be rewritten into `getByRole`?

```suggestion
				.getByRole( 'link', { name: 'Image' } )
```

Reply:
I remember this didn't work for some reason, but I can't recall it now. Will doublecheck.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/47919#:~:text=Use%20getByRole(%20%27link%27%20)%20instead%20of%20locator(%20%27a%27%20).